### PR TITLE
NUnit 3 console should produce xml events for ITestEventListener which contain unique id in the scope of all test agents for NUnit 2 tests

### PR DESCRIPTION
**Pull request by NikolayPianikov**
Use the current process id as a prefix for ID

### CODE REVIEW

**Comment by CharliePoole** [comment]
I see two problems with using the process id as the first part of the test id:

1. Although unlikely, it's theoretically possible for the process id to clash with the id used for an NUnit3 test.

2. When multiple assemblies are run in the same process, they will all have the same id.

I think this should be fixed by using the same approach as the NUnit 3 driver. Each driver takes on the unique test package ID, stored in the package. This will be unique across all packages and subpackages created for a run. See the nunit3 driver code.

**Comment by NikolayPianikov** [comment]
I've done something like that you were talking about. But I am not sure that code looks fine. I mean that now there is some implicit rule - in what order each method should be called. Also there is another rule - `TestPackage.ID` should be a string representation of an integer value because it is used as `runnerId` to call the method `RemoteTestRunner.CreateInstance(_testDomain, runnerId)`. RemoteTestRunner and ProxyTestRunner are located in NUnit.Core.dll. This assembly is used from the "lib" directory as a binary dependency and I think we should not change it.

File: src/NUnitEngine/Addins/nunit.v2.driver/NUnit2FrameworkDriver.cs [hunk 1]
```diff
@@ -68,12 +66,28 @@ public NUnit2FrameworkDriver(AppDomain testDomain)
...
 
+        private TestRunner Runner {
+            get
+            {
+                if (_runner == null)
+                {
+                    int runnerId;
+                    if (ID == null || !int.TryParse(ID, out runnerId))
+                    {
+                        throw new InvalidSuiteException("ID must be defined at first as a string representation of an integer value.");
```
Comment (CharliePoole) [inline src/NUnitEngine/Addins/nunit.v2.driver/NUnit2FrameworkDriver.cs]: This is an NUnit V2 Exception, completely unknown to the engine. Let's use NUnitEngineException, which gets special handling by some runners.

**Comment by CharliePoole** [comment]
"Implicit rule - in what order each method should be called"

Do you mean that Load must be called first? That has always been the rule. Or is it something else?

"TestPackage.ID should  be a string representation of an integer value"

That's true. But the only way to eliminate the problem is to limit NUnit V3 forever to using integer values. NUnit V2 uses ints and we can't change that unless the driver intercepts every event and modifies the id using a lookup table. I don't think we want to do that! So I think we are stuck with this bit of awkwardness so long as we are supporting V2.

File: src/NUnitEngine/Addins/nunit.v2.driver/NUnit2FrameworkDriver.cs (hunk 1 above)
Comment (CharliePoole) [inline src/NUnitEngine/Addins/nunit.v2.driver/NUnit2FrameworkDriver.cs]: Also, can we add something to the message indicating that it's the V2 driver that requires an int?

**Comment by CharliePoole** [comment]
@rprouse This looks good to merge, modulo my comments.

**Comment by NikolayPianikov** [comment]
@CharliePoole Yes I mean Load must be called first

**Comment by CharliePoole** [comment]
@NikolayPianikov The DirectRunner takes care that this is done. It probably should be documented somewhere as an assumption that a driver can make. Maybe in the interface comments?

**Comment by CharliePoole** [comment]
@rprouse Ready to merge, I think.

**Comment by NikolayPianikov** [comment]
@CharliePoole comments will never be superfluous

**Comment by rprouse** [comment]
Looks good.

**Comment by NikolayPianikov** [comment]
@CharliePoole @rprouse 

https://github.com/nunit/nunit-console/issues/30 does not work again

**Pull request merged**